src/components/HeaderSearch/index.tsx

```diff
@@ -100,7 +105,7 @@ export default function HeaderSearch (props: HeaderSearchProps) {
         onChange={onChangeInput}
       >
         <Input
-          ref={textInput}
+          ref={ref => textInput}
```

Comment:
why we need to write this like that?

Reply:
this is typecast of Ref<T> to Ref<Input>.